usecases/schema/tenant.go

```diff
@@ -169,45 +172,60 @@ func (h *Handler) DeleteTenants(ctx context.Context, principal *models.Principal
 // GetTenants is used to get tenants of a class.
 //
 // Class must exist and has partitioning enabled
-func (h *Handler) GetTenants(ctx context.Context, principal *models.Principal, class string) ([]*models.Tenant, error) {
+func (h *Handler) GetTenants(ctx context.Context, principal *models.Principal, class string,
+	after *string, limit *int64,
+) ([]*models.Tenant, error) {
 	if err := h.Authorizer.Authorize(principal, "get", tenantsPath); err != nil {
 		return nil, err
 	}
-	return h.getTenants(class)
+	return h.getTenants(class, after, limit)
 }
 
-func (h *Handler) GetConsistentTenants(ctx context.Context, principal *models.Principal, class string, consistency bool, tenants []string) ([]*models.Tenant, error) {
+func (h *Handler) GetConsistentTenants(ctx context.Context, principal *models.Principal, class string, consistency bool,
+	tenants []string, after *string, limit *int64,
+) ([]*models.Tenant, error) {
 	if err := h.Authorizer.Authorize(principal, "get", tenantsPath); err != nil {
 		return nil, err
 	}
 
 	if consistency {
-		tenants, _, err := h.metaWriter.QueryTenants(class, tenants)
+		tenants, _, err := h.metaWriter.QueryTenants(class, tenants, after, limit)
 		return tenants, err
 	}
 
 	// If non consistent, fallback to the default implementation
-	return h.getTenantsByNames(class, tenants)
+	return h.getTenantsByNames(class, tenants, after, limit)
 }
 
-func (h *Handler) getTenants(class string) ([]*models.Tenant, error) {
+func (h *Handler) getTenants(class string, after *string, limit *int64) ([]*models.Tenant, error) {
 	info, err := h.multiTenancy(class)
 	if err != nil || info.Tenants == 0 {
 		return nil, err
 	}
 
 	ts := make([]*models.Tenant, info.Tenants)
 	f := func(_ *models.Class, ss *sharding.State) error {
-		if n := len(ss.Physical); n > len(ts) {
-			ts = make([]*models.Tenant, n)
-		} else if n < len(ts) {
-			ts = ts[:n]
+		if after != nil && limit != nil {
```

Comment:
This code is similar to the `cluster/store/schema.go:getTenants`, can it be changed to avoid duplicating the logic (eg "DRY'd up"?)